kogito-benchmarks-framework/src/main/java/org/kie/kogito/benchmarks/framework/Commands.java

```diff
@@ -0,0 +1,484 @@
+            Assertions.assertThat(newCPUAffinity).contains(expectedAffinityList);
+            p.waitFor();
+        }
+        Assertions.assertThat(p.exitValue()).isEqualTo(0);
+    }
+
+    public static void cleanDirOrFile(String... path) {
+        for (String s : path) {
+            try {
+                Files.walk(Paths.get(s))
+                        .sorted(Comparator.reverseOrder())
+                        .map(Path::toFile)
+                        .forEach(File::delete);
+            } catch (IOException e) {
+                logger.warn("Unable to delete directories or files", e);
+            }
+        }
+    }
+
+    public static List<String> getRunCommand(String[] baseCommand) {
+        List<String> runCmd = new ArrayList<>();
+        if (isThisWindows) {
+            runCmd.add("cmd");
+            runCmd.add("/C");
+        }
+        runCmd.addAll(Arrays.asList(baseCommand));
+
+        return Collections.unmodifiableList(runCmd);
+    }
+
+    public static List<String> getBuildCommand(String[] baseCommand) {
+        List<String> buildCmd = new ArrayList<>();
+        if (isThisWindows) {
+            buildCmd.add("cmd");
+            buildCmd.add("/C");
+        }
+        buildCmd.addAll(Arrays.asList(baseCommand));
+        buildCmd.add("-Dmaven.repo.local=" + getLocalMavenRepoDir());
+        // use the same settings.xml as the main build
+        buildCmd.add("-s");
+        buildCmd.add(System.getProperty("maven.settings"));
+
+        return Collections.unmodifiableList(buildCmd);
+    }
+
+    public static boolean waitForTcpClosed(String host, int port, long loopTimeoutS) throws InterruptedException, UnknownHostException {
+        InetAddress address = InetAddress.getByName(host);
+        long now = System.currentTimeMillis();
+        long startTime = now;
+        InetSocketAddress socketAddr = new InetSocketAddress(address, port);
+        while (now - startTime < 1000 * loopTimeoutS) {
+            try (Socket socket = new Socket()) {
+                // If it let's you write something there, it is still ready.
+                socket.connect(socketAddr, 1000);
+                socket.setSendBufferSize(1);
+                socket.getOutputStream().write(1);
+                socket.shutdownInput();
+                socket.shutdownOutput();
+                logger.info("Socket still available: " + host + ":" + port);
+            } catch (IOException e) {
+                // Exception thrown - socket is likely closed.
+                return true;
+            }
+            Thread.sleep(1000);
+            now = System.currentTimeMillis();
+        }
+        return false;
+    }
+
+    // TODO this may be useful in the future
+    public static void adjustPrettyPrintForJsonLogging(String appDir) throws IOException {
+        Path appProps = Paths.get(appDir + File.separator + "src" + File.separator + "main" + File.separator + "resources" + File.separator + "application.properties");
```

Comment:
`Path.of()` instead of `Paths.get()` ?

Reply:
Actually `Paths.get()` can be used as well, but not with a single string, probably a mistake by a previous author. FYI I also found that `Paths` Javadoc states since Java 11 that the class may be deprecated in the future and a user should use `Path.of()`.